client/web/src/auth/welcome/useHubSpotForm.tsx

```
@@ -0,0 +1,109 @@
+    onFormSubmitted?: () => void
+}
+
+export function useHubSpotForm(config: HubSpotConfig): React.ReactNode {
+    const [isScriptLoaded, setIsScriptLoaded] = useState<boolean>(false)
+    const [isFormRendered, setIsFormRendered] = useState<boolean>(false)
+    const [containerId] = useState(() => uuid.v4())
+
+    const onLoad = useCallback(() => {
+        setIsScriptLoaded(true)
+    }, [])
+    const onError = useCallback(() => {
+        throw new Error('Failed to load HubSpot form')
+    }, [])
+
+    const { onFormSubmitted, formId } = config
+    const onFormSubmittedCallback = useCallback(
+        event => {
+            if (typeof onFormSubmitted !== 'function') {
+                return
+            }
+
+            if (
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                event.data.type === 'hsFormCallback' &&
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                event.data.eventName === 'onFormSubmit' &&
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                event.data.id === formId
+            ) {
+                onFormSubmitted()
+            }
+        },
+        [onFormSubmitted, formId]
+    )
+
+    useEffect(() => {
+        if (globalHubSpotScriptInsertedPromise !== null) {
+            // If the script was already added by any other callers of the hook, we don't have to
+            // load it again.
+            globalHubSpotScriptInsertedPromise.then(onLoad, onError)
+            return
+        }
+        globalHubSpotScriptInsertedPromise = new Promise((resolve, reject) => {
+            const script = document.createElement('script')
+            script.src = HUBSPOT_EXTERNAL_SCRIPT
+            script.async = true
+            script.addEventListener('load', () => resolve())
+            script.addEventListener('error', () => reject(new Error('Error loading script tag')))
```

I think this error message is just swallowed by `onError()`. I see no way to use it (e.g. log it or print it).
Suggested solution: in `onError()`, include the original error message, e.g. throw `Failed to load HubSpot form: {original error message}`.